### PR TITLE
feat(web): optimization via lazy preprocessing of keyboard touch-layout info ⏩

### PR DESCRIPTION
**Pull request by jahorton**
Follows #11264, addressing part of #11166. 

How can we go "further beyond" with OSK loading speed than not building all of its layers initially?  Why, by not even _pre-processing_ all of its layers initially!  This PR extends the just-in-time (JIT) behavior from the prior PR by only preprocessing layer-specifications as they're needed.

So, revisiting that `gff_amharic` stress-test from #11263, performed on a SM-T350 / Galaxy Tab A...

gff_amharic "before"

Loading time "before": between 3.5 to 4 seconds.

gff_amharic "after"

Loading time "after": 1.2 seconds, with _half_ of that being keyboard-script load.  

To be clear, underneath the visible `buildLayer` and `getLayer` stackframes in the second ("after") image, the visible `s...e` is `sanitize`, with the sliver to its right being `polyfill`.  This is due to the newly-introduced JIT pattern - both are executing for a single layer, and only once it is needed.

Actual time spent preprocessing & building the OSK element hierarchy:  just **_120ms_**.  Remember, "before", this section cost us 2.66 seconds!  Of course, we'll have to pay that 2.66 seconds over time _as_ we visit every layer... assuming we actually _do_ visit every layer.

Clearly, the easiest way to optimize `sanitize` is to just... not do it until we have to.  One layer isn't _that_ expensive, even on older devices.  42 of them?  Now that... _that_ starts to add up.  Of course, this does mean that each swap to a previously-unused layer will be a bit slower than before.

----

Other keyboards:
- `sil_euro_latin` now loads in approximately 764ms, down from 1.1 second as seen in https://github.com/keymanapp/keyman/pull/11140#issuecomment-2036287502.
    - `sil_euro_latin` automatically swaps to `shift` with a new context, which was the case for this test.  So, it built an extra layer "out of the gate" compared to the other two below.
    - This is also swapping out of `gff_amharic`, which featured an active lexical model.
- `sil_cameroon_azerty` - 629 ms.
- `khmer_angkor` - 689 ms.

So, we're going somewhere around 50% faster (or better!) for keyboard loading with more "conventional" keyboards than we were with #11140, which was _already_ a marked improvement on what came before.

## User Testing

TEST_ANDROID_USE:  Test the Keyman for Android app using `sil_euro_latin``.
- Forcibly kill and restart the app at least three times during this test.
- Be sure to use a key from every layer during this test.
- Use at least one longpress, one downflick, and one layer-swapping multitap.

TEST_IOS_USE:  Test the Keyman for iPhone and iPad app using `sil_euro_latin`.
- Forcibly kill and restart the app at least three times during this test.
- Be sure to use a key from every layer during this test.
- Use at least one longpress, one downflick, and one layer-swapping multitap.

### CODE REVIEW

**Comment by dinakaranr** [comment]
# Test Results

* **TEST_ANDROID_USE (PASSED):** I tested this issue with the attached "Keyman-18.0.59-alpha-test-11265" build on the Android 14(Physical devices). Here is my observation.

1. Installed the "Keyman-18.0.59.apk" file and gave all permissions to the application.
2. Checked the "Enable Keyman as system-wide keyboard" and set the keyboard as the default keyboard box on the settings page.
3. Open the Keyman app. Enable the "Predictions" and install the "Dictionary."
4. Select the "sil_euro_latin" keyboard.
5. Enter some letters from the QWERTY key, sub-menu letters, long-press, double tap(cap lock), down-flick and special characters (swapping multitap).
6. Selected some text from every layer.
7. Kill the apps multiple times.
8. Keyboards are responsible and give input when reopening the app. Thank you. 

* **TEST_IOS_USE (PASSED):**  I tested this issue with the attached "Keyman-18.0.59-alpha-test-11265" build on the iphone 13(Physical devices).
I followed the steps that are mentioned on Android. 
Observed that the Keyboards are responsible and give input when reopening the app. they perform good when adding text from mulitiple layer and killing the app.
It works well. Thank you.

File: common/web/keyboard-processor/src/keyboards/activeLayout.ts [hunk 1]
```diff
@@ -778,7 +768,11 @@ export class ActiveLayer implements LayoutLayer {
 }
 
 export class ActiveLayout implements LayoutFormFactor{
-  layer: ActiveLayer[];
+  /**
+   * Holds all layer specifications for the layout.  There is no guarantee that they
+   * have been fully preprocessed.
+   */
+  layer: TouchLayerSpec[];
```
Comment (ermshiperete) [inline common/web/keyboard-processor/src/keyboards/activeLayout.ts:775]: Rename?

```suggestion
  layers: TouchLayerSpec[];
```

**Review by ermshiperete — APPROVED** [review]

File: common/web/keyboard-processor/src/keyboards/activeLayout.ts (hunk 1 above)
Comment (jahorton) [reply common/web/keyboard-processor/src/keyboards/activeLayout.ts:775]: So... this is a request to refactor the variable to be plural?  There's no bug at the moment - the variable is currently `layer`, not `layers`.

Comment (jahorton) [reply common/web/keyboard-processor/src/keyboards/activeLayout.ts:775]: If we were to do this, it'd likely be best as a separate PR, just to keep things extra clear.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.61-alpha](https://keyman.com/downloads/releases/alpha/18.0.61)

**Pull request merged**